### PR TITLE
Fix bug and add unit test for dataloader v2

### PR DESCRIPTION
**Pull request by oscarwumit**
## Description
1. Fixed a bug where wrong variables were assigned when unpacking Datum. Specifically, `lt_masks` should have higher precedence than `gt_masks` based on the definition in Datum class.
2. Added initial unit tests for dataloader in chemprop v2. This improves coverage from 0% to 62%. Will add more in later commits.

### CODE REVIEW

**Comment by davidegraff** [comment]
can you rewrite your test case using either `pytest.mark.parameterize` or `pytest.fixture`?

**Comment by oscarwumit** [comment]
Thanks for the comment, I have updated the tests to use `pytest.fixture`

File: tests/unit/data/test_dataloader.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    mgs, V_ds, x_fs, ys, weights, lt_masks, gt_masks = result
+    
+    assert isinstance(result, tuple)
+    assert isinstance(mgs, BatchMolGraph)
+    assert torch.equal(V_ds, torch.tensor([1.0, 2.0], dtype=torch.float32))
+    assert torch.equal(x_fs, torch.tensor([[3.0, 4.0]], dtype=torch.float32))
+    assert torch.equal(ys, torch.tensor([[6.0, 7.0]], dtype=torch.float32))
+    assert torch.equal(weights, torch.tensor([[[8.0]]], dtype=torch.float32))
+    assert torch.equal(lt_masks, torch.tensor([[9.0]], dtype=torch.float32))
+    assert torch.equal(gt_masks, torch.tensor([[10.0]], dtype=torch.float32))
```
Comment (davidegraff) [inline tests/unit/data/test_dataloader.py:72]: use `torch.testing.assert_close` for these

File: tests/unit/data/test_dataloader.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    mgs, V_ds, x_fs, ys, weights, lt_masks, gt_masks = result
+    
+    assert isinstance(result, tuple)
+    assert isinstance(mgs, BatchMolGraph)
+    assert torch.equal(V_ds, torch.tensor([1.0, 2.0, 5.0, 7.0], dtype=torch.float32))
+    assert torch.equal(x_fs, torch.tensor([[3.0, 4.0], [8.0, 9.0]], dtype=torch.float32))
+    assert torch.equal(ys, torch.tensor([[6.0, 7.0], [6.0, 4.0]], dtype=torch.float32))
+    assert torch.equal(weights, torch.tensor([[[8.0]], [[1.0]]], dtype=torch.float32))
+    assert torch.equal(lt_masks, torch.tensor([[9.0], [2.0]], dtype=torch.float32)) 
+    assert torch.equal(gt_masks, torch.tensor([[10.0], [3.0]], dtype=torch.float32))
```
Comment (davidegraff) [inline tests/unit/data/test_dataloader.py:89]: same here

File: tests/unit/data/test_dataloader.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+
+def test_collate_batch_multiple_graphs(datum_1, datum_2):
+    
+    batch = [datum_1, datum_2]
+    
+    result = collate_batch(batch)
+    mgs, V_ds, x_fs, ys, weights, lt_masks, gt_masks = result
+    
+    assert isinstance(result, tuple)
```
Comment (davidegraff) [inline tests/unit/data/test_dataloader.py:82]: this is implicitly tested above

File: tests/unit/data/test_dataloader.py (hunk 1 above)
Comment (am2145) [reply tests/unit/data/test_dataloader.py]: I have updated the test to use torch.testing.assert_close as requested.

File: tests/unit/data/test_dataloader.py (hunk 2 above)
Comment (am2145) [reply tests/unit/data/test_dataloader.py]: As above, this has been updated to use torch.testing.assert_close.

File: tests/unit/data/test_dataloader.py (hunk 3 above)
Comment (am2145) [reply tests/unit/data/test_dataloader.py]: This assertion has been removed.

**Comment by am2145** [comment]
I have updated test_dataloader.py and moved it to the v2 tests folder to keep it with the other data tests. Aside from the changes requested by David, the weight in the test is edited to be a float, and `lt_mask` and `gt_mask` were changed to be Booleans in line with their application elsewhere.

**Pull request merged**